### PR TITLE
Enable and apply isort

### PR DESCRIPTION
**Pull request by alexfikl**
Mostly follows inducer/pytools#169.

### CODE REVIEW

File: boxtree/__init__.py [hunk 1]
```diff
@@ -20,12 +20,16 @@
...
-    refine_and_coarsen_tree_of_boxes, make_meshmode_mesh_from_leaves)
+from boxtree.tree import (Tree, TreeOfBoxes, TreeWithLinkedPointSources,
+                          box_flags_enum)
 from boxtree.tree_build import TreeBuilder
+from boxtree.tree_of_boxes import (coarsen_tree_of_boxes,
+                                   make_meshmode_mesh_from_leaves,
+                                   make_tree_of_boxes_root,
+                                   refine_and_coarsen_tree_of_boxes,
+                                   refine_tree_of_boxes,
+                                   uniformly_refine_tree_of_boxes)
```
Comment (alexfikl) [inline boxtree/__init__.py:31]: `isort` seems to have some options for how [multiline imports are handled](https://pycqa.github.io/isort/docs/configuration/multi_line_output_modes.html)?

Comment (inducer) [reply boxtree/__init__.py:31]: Got a preference?

Comment (alexfikl) [reply boxtree/__init__.py:31]: Probably "4 - Hanging Grid"? I like my horizontal space :grin: 
```
from third_party import (
    lib1, lib2, lib3, lib4,
    lib5, ...)
```

Comment (inducer) [reply boxtree/__init__.py:31]: They have this "centralized config" thing, but... I'm thinking that's overkill? What do you think?

Comment (alexfikl) [reply boxtree/__init__.py]: Where's that described? Do you mean the "profiles"?

**Comment by inducer** [comment]
Thx!

File: boxtree/__init__.py (hunk 1 above)
Comment (alexfikl) [reply boxtree/__init__.py]: Yeah, it seems like they're hardcoded, so probably overkill for now.

**Comment by inducer** [comment]
How do you feel about bf262f4e32976bc8743e88058fc1682937fe6468?

**Pull request merged**